File: 728-SelfDividingNumbers/728-SelfDividingNumbers.js
// Last updated: 3/13/2026, 10:14:28 AM
1/**
2 * @param {number} left
3 * @param {number} right
4 * @return {number[]}
5 */
6var selfDividingNumbers = function(left, right) {
7    let res = [];
8
9    for(let i=left;i<=right;i++){
10        let num = i;
11
12        while(num>0){
13            let remainder = num % 10;
14
15            if(remainder === 0 || i%remainder != 0){
16                break;
17            }
18            num = Math.floor(num/10);
19        }
20
21        if(num===0){
22            res.push(i);
23        }
24    }
25    return res;
26};

File: 728-SelfDividingNumbers/728-SelfDividingNumbers.cpp
// Last updated: 5/21/2026, 10:10:25 AM
1class Solution {
2public:
3    vector<int> selfDividingNumbers(int left, int right) {
4        vector<int> res;
5
6        for(int i=left;i<=right;i++){
7            int num = i;
8
9            while(num>0){
10                int remainder = num % 10;
11
12                if(remainder == 0 || i % remainder != 0){
13                    break;
14                }
15
16                num = num / 10;
17            }
18
19            if(num == 0){
20                res.push_back(i);
21            }
22        }
23        return res;
24    }
25};